### PR TITLE
LPS-127874 The sidebar does not lose the focus on the field settings when clicking away of the field

### PR DESCRIPTION
**Pull request by boton**
Based on https://github.com/liferay-echo/liferay-portal/pull/3733

**Test Plan**
1. Go to Document types in document and media
2. Click in add a new document type
3. Add any field in the form
4. Click outside the field

**Expected result:**
The sidebar loses focus on the field.

**Actual result:**
The sidebar focus is still in the last selected field.

---

**After the fix:**

https://user-images.githubusercontent.com/67901/108036349-63379a80-7038-11eb-9a30-2dfd6b5126ad.mp4

### CODE REVIEW

**Comment by boton** [comment]
ci:test:relevant

**Comment by boton** [comment]
ci:test:sf

**Comment by boton** [comment]
Update wrong LPS :D

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/data-engine/DataEngineLayoutBuilderHandler.es.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+					dataDefinition: JSON.stringify(dataDefinition),
+					dataLayout: JSON.stringify(dataLayout),
+				},
+			});
+		});
+	};
+
+	form.addEventListener('submit', saveDataEngineStructure);
+
+	// Deselect field when clicking outside the form builder
```
Comment (4lejandrito) [inline modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/data-engine/DataEngineLayoutBuilderHandler.es.js:89]: Is this comment necessary?

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/data-engine/DataEngineLayoutBuilderHandler.es.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+export default function ({defaultLanguageId, namespace}) {
+	const form = document.getElementById(`${namespace}fm`);
+
+	const getDataLayoutBuilder = () => {
+		return Liferay.componentReady(`${namespace}dataLayoutBuilder`);
+	};
+
+	const getInputLocalizedValues = (fieldName) => {
+		const inputLocalized = Liferay.component(`${namespace}${fieldName}`);
```
[the remark this answers is not in the document]
Comment (boton) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/data-engine/DataEngineLayoutBuilderHandler.es.js:23]: I think is safe with the IF after avoiding errors only return an empty object if isn´t create

[the remark this answers is not in the document]
Comment (boton) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/data-engine/DataEngineLayoutBuilderHandler.es.js:23]: makes sense to you?

[the remark this answers is not in the document]
Comment (carloslancha) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/data-engine/DataEngineLayoutBuilderHandler.es.js:23]: 👍

**Comment by carloslancha** [comment]
LGTM

**Review by carloslancha — APPROVED** [review]

**Review by 4lejandrito — APPROVED** [review]

**Comment by boton** [comment]
ci:forward

**Pull request closed without merging**